docs/custom-account-fields.md

```diff
@@ -0,0 +1,182 @@
+# Account custom fields
+
+This document shows you how to setup account custom fields for your galoy instance.
+
+## Configuration
+
+```
+customFields:
+  - name: "customFieldName"
+    type: <"string" | "integer" | "float" | "boolean"> # optional (default: "string")
+    defaultValue: <value> # optional
+    editable: <true | false> # optional (default: false) - true if field can be updated by the user
+    required: <true | false> # optional (default: false) - true if field is required by mutations
+    index: <true | false> # optional (default: false) - true if field should have an index in db
+    unique: <true | false> # optional (default: false) - true if field should have a unique index in db
+```
+
+## Example
+
+Add the next configuration to `default.yaml`
+```
+accounts:
+  ...
+  customFields:
+    - name: "nationalId"
+      type: "integer"
+      defaultValue: 0
+    - name: "firstName"
+      type: "string"
+      required: true
+      editable: true
+    - name: "lastName"
+      type: "string"
+      editable: true
+    - name: "terms"
+      type: "boolean"
+      editable: true
+      defaultValue: false
+```
+
+If customFields config is empty then all related mutations and queries should not be visible by API clients.
+
+### Update values as an user
+
+In the main server playground (`make start-main`) run the next:
+
+```
+mutation AccountCustomFieldsUpdate($input: AccountCustomFieldsUpdateInput!) {
+  accountCustomFieldsUpdate(input: $input) {
+    errors {
+      message
+    }
+    accountCustomFields {
+      firstName
+      lastName
+      terms
+    }
+  }
+}
+```
+Input
+```
+{"input": {  "firstName": "Satoshi", "lastName": "Nakamoto", "terms": true } }
+```
+
+Note: please notice that `nationalId` can only be updated as an editor (admin API)
+
+### Query values as an user
+
+```
+query me {
+  me {
+    defaultAccount {
+      defaultWalletId
+      wallets {
+        id
+        walletCurrency
+      }
+      customFields {
+        firstName
+        lastName
+        terms
+      }
+    }
+  }
+}
+```
+
+Note: please notice that `nationalId` is only visible as an editor (admin API)
```

Comment:
I think the `editable` is conflating multiple use case. 
- it's possible that some fields could be write-once from the end user. Ie: the nationalId could be something submitted by the end user. 
-- similarly, the firstName is (in my mind) write-once by the user, but should not be editable.

- a field, even if not editable, could be still visible by the end user (I'm not sure about that use case, trying to think about it?)


The admin should be able to see/edit everything IMO, at least for now.

Reply:
if we had a `visible` field, then this could be the flag on whether the mutation and query would be present or not for the end user API.

the `editable` field would then only be relevant if `visible: true` and if yes, means that the field is write only, if no, means that the fields can be write several time

wdyt?

Reply:
could also be `viewable` instead of `visible`

Reply:
if `visible` and `editable` are connected - perhaps there is a nested structure:
```
expose_to_user:
  visible: true
  editable: false
```

?